maxim/logger/fireworks/client.py

```
@@ -0,0 +1,223 @@
+    This function instruments the Fireworks SDK by patching the chat completion
+    methods to automatically log API calls, model parameters, and responses to
+    Maxim. It supports both synchronous and asynchronous operations, streaming
+    responses, and various Fireworks specific features.
+    """
+
+    global _INSTRUMENTED
+    if _INSTRUMENTED:
+        scribe().info("[MaximSDK] Fireworks already instrumented")
+        return
+
+    def wrap_create(create_func):
```

Ruff (0.12.2)

`wrap_create` is too complex (15 > 10)

(C901)